packages/runtime/src/helpers/config.ts

```diff
@@ -96,6 +96,12 @@ export const hasManuallyAddedModule = ({
   )
 /* eslint-enable camelcase */
 
+/**
+ * Transforms `/api/shows/[id].js` into `/api/shows/*id*.js`,
+ * so that the file `[id].js` is matched correctly.
+ */
+const escapeGlob = (path: string) => path.replace(/\[/g, '*').replace(/]/g, '*')
```

Comment:
i think this is going to match unwanted files, e.g. `grid.js`
is it possible to escape the path, e.g. `\[id\].js` ?

Reply:
I tried escaping the path, and glob patterns do allow escaping the brackets - but I couldn't get that to work properly with how ZISI interprets the globs. Maybe we could open a follow-up issue to investigate this.

Yes, this matches too many files - but I don't think that it'll be a big problem in practice.

Reply:
Good to know it's fixed in v9! I'd prefer the follow-up. Let's open an issue for this.

Reply:
Opened https://github.com/netlify/zip-it-and-ship-it/issues/1459